prov/efa/src/rdm/rxr_msg.c

```diff
@@ -1432,10 +1417,10 @@ ssize_t rxr_msg_trecvmsg(struct fid_ep *ep_fid, const struct fi_msg_tagged *tmsg
 	ep = container_of(ep_fid, struct rxr_ep, base_ep.util_ep.ep_fid.fid);
 
 	if (flags & FI_PEEK) {
-		ret = rxr_msg_peek_trecv(ep_fid, tmsg, flags);
+		ret = rxr_msg_peek_trecv(ep_fid, tmsg, flags | ep->base_ep.util_ep.rx_msg_flags);
```

Comment:
This looks fishy though. Why are we not respecting application flags? Maybe comment the reason here.

Reply:
See the commit https://github.com/ofiwg/libfabric/commit/978ec0261d0a29b0e60a0de9ebb7b48f528e69ee

```
rxr_msg_recv (trecv, recvv, trecvv) should pass flags as
rxr_rx_flags(ep) to rxr_msg_generic_recv, which means it should
enable FI_COMPLETION in the flags as long as the prov_info->rx_attr.flags
support FI_COMPLETION. For rxr_msg_recvmsg (trecvmsg), it should
pass application flags |= util_ep.rx_msg_flags, which will have NO FI_COMPLETION
when application binds rx cq with FI_SELECTIVE_COMPLETION, and does not have
FI_COMPLETION in the flags of fi_recvmsg.
```

I can add comment in the code

Reply:
This is how other providers (like shm) do as well

Reply:
Updated

Reply:
Thanks. This is better. It was odd that L1241 & L1302 were comment fine but not here.